presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveQueries.java

```diff
@@ -811,6 +811,12 @@ public void testRow()
         assertQuery("SELECT cast(row(nationkey, regionkey) as row(a bigint, b bigint)) FROM nation");
     }
 
+    @Test
+    public void testNullConstantMap()
+    {
+        assertQuery("SELECT if((col_str='foo'), col_map, null) FROM (VALUES ('foo', MAP(ARRAY['bar'], ARRAY[123.456]))) AS t(col_str, col_map)");
```

Comment:
nit: orders_ex has a map column, hence, can be used to write a bit more realistic query

```
SELECT if(orderkey % 2 = 0, quantity_by_linenumber) FROM order_ex
```